src/main/java/fr/lunatech/timekeeper/services/EventTemplateService.java

```diff
@@ -64,38 +66,101 @@
+                eventTemplate.name,
+                eventTemplate.startDateTime,
+                eventTemplate.endDateTime, ctx);
+
+        if (checkEvent) {
+            return Optional.empty();
+        }
+
         eventTemplate.persistAndFlush();
-        return eventTemplate.id;
+        return Optional.of(eventTemplate.id);
     }
 
     @Transactional
-    public Optional<Long> update(Long id, EventTemplateRequest request, AuthenticationContext ctx){
-        logger.debug("Modify event template for for id={} with {}, {}", id, request, ctx);
+    public Optional<Long> update(Long eventId, EventTemplateRequest request, AuthenticationContext ctx) {
+        logger.debug("Modify event template for for id={} with {}, {}", eventId, request, ctx);
+
+        // Request coming from frontend
+        final var newName = request.getName();
+        final var newStartTime = request.getStartDateTime();
+        final var newEndTime = request.getEndDateTime();
+
         // early quit: eventTemplate doesn't exist
-        final Optional<EventTemplate> maybeEvent = findById(id,ctx);
-        if (maybeEvent.isEmpty()){
-            return Optional.empty();
-        }
-        // delete all userEvent associated to the previous state of this event template
-        deleteAttendees(maybeEvent.get());
+        final Optional<EventTemplate> maybeEvent = findEventById(eventId, ctx);
+
+        return maybeEvent.map(event -> {
+            boolean checkEvent = validateEventUpdation(
+                    newName,
+                    event.id,
+                    newStartTime,
+                    newEndTime,
+                    ctx);
+
+            if (checkEvent) {
+                throw new UpdateResourceException(
+                        "Event with name: " + request.getName() +
+                        ", already exists with same Start: " + request.getStartDateTime().toLocalDate() +
+                        "and End: " + request.getEndDateTime().toLocalDate() + " dates."
+                );
+            }
+            // delete all userEvent associated to the previous state of this event template
+            deleteAttendees(event);
+
+            // actually update the event. by side effect every userEvent associated will be created
+            EventTemplate eventTemplateUpdated = request.unbind(event, userService::findUserById, ctx);
+            eventTemplateUpdated.persistAndFlush();
+
+            return eventTemplateUpdated.id;
+        });
+    }
 
-        // actually update the event. by side effect every userEvent associated will be created
-        EventTemplate eventTemplateUpdated = request.unbind(maybeEvent.get(),userService::findById, ctx);
-        eventTemplateUpdated.persistAndFlush();
 
-        return Optional.of(eventTemplateUpdated.id);
+    private boolean validateEventUpdation(String name, Long eventId, LocalDateTime startDateTime, LocalDateTime endDateTime, AuthenticationContext ctx) {
```

Comment:
Updation isn't a correct word
source : https://english.stackexchange.com/questions/68169/is-updation-a-correct-word